backends/xsmm/ceed-xsmm-tensor.c

```diff
@@ -75,8 +75,10 @@ static int CeedTensorContractApply_Xsmm(CeedTensorContract contract, CeedInt A,
   ierr = CeedTensorContractGetData(contract, (void *)&impl); CeedChk(ierr);
 
   // Get nelem and current dim
-  CeedScalar currdim = log(C/blksize) / log(J);
-  if (!(C % blksize) && currdim - (int)currdim < 1e-15) {
+  CeedScalar currdim = log(C/blksize);
+  if (fabs(log(J)) > CEED_EPSILON)
+    currdim /= log(J);
+  if (!(C % blksize) && currdim - (int)currdim < CEED_EPSILON) {
```

Comment:
Surely this calculation can be done purely with integer arithmetic.

Reply:
Maybe. I need to get `currdim` out of the 4 numbers `(ncomp*P**(dim-1-currdim), P, nelem*Q**currdim, Q)` without knowing what `ncomp` or `nelem` are.